### PR TITLE
Para sub

### PR DESCRIPTION
**Pull request by SaMotlagh**
1. convert all consensus storage to doubleStorageMap.
2. update registration process accordingly (Register one neuron on the requested subnetwork).

### CODE REVIEW

File: pallets/paratensor/src/epoch.rs [hunk 1]
```diff
@@ -163,17 +165,17 @@ impl<T: Config> Pallet<T> {
     pub fn vec_u16_proportions_to_fixed( vec: Vec<u16> ) -> Vec<I32F32> { vec.into_iter().map(|e| Self::u16_proportion_to_fixed(e) ).collect() }
     pub fn vec_fixed_proportions_to_u16( vec: Vec<I32F32> ) -> Vec<u16> { vec.into_iter().map(|e| Self::fixed_proportion_to_u16(e) ).collect() }
 
-    pub fn set_ranks( netuid:u16, ranks:Vec<I32F32> ) { Rank::<T>::insert( netuid, Self::vec_fixed_proportions_to_u16( ranks ) ) }
-    pub fn set_trust( netuid:u16, trust:Vec<I32F32> ) { Trust::<T>::insert( netuid, Self::vec_fixed_proportions_to_u16( trust ) ) }
-    pub fn set_consensus( netuid:u16, consensus:Vec<I32F32> ) { Consensus::<T>::insert( netuid, Self::vec_fixed_proportions_to_u16( consensus ) ) }
-    pub fn set_incentives( netuid:u16, incentive:Vec<I32F32> ) { Incentive::<T>::insert( netuid, Self::vec_fixed_proportions_to_u16( incentive ) ) }
-    pub fn set_dividends( netuid:u16, dividends:Vec<I32F32> ) { Dividends::<T>::insert( netuid, Self::vec_fixed_proportions_to_u16( dividends ) ) }
+    pub fn set_ranks( netuid:u16, neuron_uid: u16, ranks:u16 ) { Rank::<T>::insert( netuid, neuron_uid, ranks) }
```
Comment (shibshib) [inline pallets/paratensor/src/epoch.rs:168]: Converting `Vec<I32F32>` to regular `u16` means set_ranks will no longer take a vector of ranks, correct? Just a singular rank for a singular UID?

File: pallets/paratensor/src/lib.rs [hunk 2]
```diff
@@ -1,20 +1,46 @@
...
 
 #[cfg(feature = "runtime-benchmarks")]
 mod benchmarking;
 
+/// ************************************************************
+///	-Paratensor-Imports
+/// ************************************************************
+mod registration;
+mod epoch;
+mod misc;
```
Comment (shibshib) [inline pallets/paratensor/src/lib.rs:33]: if `misc` is just a collection of miscellaneous functions, we should call this utils instead.

File: pallets/paratensor/src/misc.rs [hunk 3]
```diff
@@ -6,7 +6,7 @@ impl<T: Config> Pallet<T> {
 	/// ==== Misc ====
 	/// ==============
     pub fn get_total_issuance() -> u64 { return TotalIssuance::<T>::get() }
-    pub fn get_total_stake() -> u64 { return TotalStake::<T>::get() }
+    //pub fn get_total_stake() -> u64 { return TotalStake::<T>::get() }
```
Comment (shibshib) [inline pallets/paratensor/src/misc.rs:9]: if we're not using it anymore, let's just remove this.

File: pallets/paratensor/src/epoch.rs (hunk 1 above)
Comment (SaMotlagh) [reply pallets/paratensor/src/epoch.rs:168]: yes, bcz we converted Rank storage to doubleStorageMap that maps net_UID --> Neuron UID --> rank. This function is only defined there and have not revoked anywhere else. We might change it later.

File: pallets/paratensor/src/lib.rs (hunk 2 above)
Comment (SaMotlagh) [reply pallets/paratensor/src/lib.rs:33]: makes sense. not sure what we are doing there :) will keep it in mind when I touch it.

File: pallets/paratensor/src/misc.rs (hunk 3 above)
Comment (SaMotlagh) [reply pallets/paratensor/src/misc.rs:9]: right!

File: pallets/paratensor/src/epoch.rs (hunk 1 above)
Comment (shibshib) [reply pallets/paratensor/src/epoch.rs:168]: Understood. thanks!

**Review by shibshib — APPROVED** [review]

**Pull request merged**